Source/NSURLProtocol.m

```diff
@@ -1802,10 +1979,32 @@ - (void) stopLoading
     {
       [this->input setDelegate: nil];
       [this->output setDelegate: nil];
-      [this->input removeFromRunLoop: [NSRunLoop currentRunLoop]
-			     forMode: NSDefaultRunLoopMode];
-      [this->output removeFromRunLoop: [NSRunLoop currentRunLoop]
-			      forMode: NSDefaultRunLoopMode];
+      if (this->customOperationQueue)
```

Comment:
Could you please try to reduce the duplication of code in this class? To me it seems like you had to copy similar bits over and over again.